server/services/formService.js

```diff
@@ -766,24 +766,63 @@ module.exports = function createFormService(formClient) {
       newOffenderNo,
       transactionalClient
     )
+    logger.info(`Merge: updateOffenderIdentifierReturningBookingIdForm: rows updated = ${result1.rowCount}`)
+
     const result2 = await formClient.updateOffenderIdentifierReturningBookingIdLite(
       oldOffenderNo,
       newOffenderNo,
       transactionalClient
     )
+    logger.info(`Merge: updateOffenderIdentifierReturningBookingIdLite: rows updated = ${result2.rowCount}`)
+
     const result3 = await formClient.updateOffenderIdentifierRiskChange(
       oldOffenderNo,
       newOffenderNo,
       transactionalClient
     )
-    const result4 = await formClient.updateOffenderIdentifierSecurityReferral(
-      oldOffenderNo,
-      newOffenderNo,
-      transactionalClient
-    )
-    logger.info(
-      `Merge summary: rows updated = ${result1.rowCount} ${result2.rowCount} ${result3.rowCount} ${result4.rowCount}`
-    )
+    logger.info(`Merge: updateOffenderIdentifierRiskChange: rows updated = ${result3.rowCount}`)
+
+    const securityReferralRemove = await getSecurityReferral(oldOffenderNo, transactionalClient)
+    const securityReferralSurvives = await getSecurityReferral(newOffenderNo, transactionalClient)
+
+    // If row for surviving offenderNo exists, or to-be-removed offenderNo exists, use that.
+    // If both exist, use whichever is not in COMPLETED status
+    // If neither COMPLETED, use the later raisedDate
+    // If both COMPLETED, do nothing.
+    // NOTE no need to delete the removed record when not used.
+
```

Comment:
This is horrible because offenderNo is a unique key and its possible (though unlikely) that both the offenderNos have been flagged, with e.g. , the one to be deleted having the 'NEW' status, so being the record we want to keep.